raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansProcessor.java

```diff
@@ -151,15 +191,21 @@ public void process(Record<TraceIdentity, RawSpan> record) {
     TraceIdentity key = record.key();
     RawSpan value = record.value();
     TraceState traceState = traceStateStore.get(key);
-    boolean firstEntry = (traceState == null);
 
     if (shouldDropSpan(key, traceState)) {
       return;
     }
 
+    Event event = value.getEvent();
+    if (isPeerServiceNameIdentificationRequired(event)) {
+      processSpanForPeerServiceNameIdentification(key, value, traceState, currentTimeMs);
+      return;
```

Comment:
should we return?
what if we just resolve the peer service and save in state store and process the spans via default behaviour? was it because of 30sec delay?